doc/source/working_with_kiwi/xml_description.rst

```diff
@@ -0,0 +1,271 @@
+sections:
+
+- `iso`: a live ISO image, see :ref:`hybrid_iso`
+- `vmx`: build a virtual machine image, see: :ref:`vmx`
+- `oem`: results in an expandable image that can be deployed via a bootable
+  installation medium, e.g. a USB drive or a CD. See :ref:`oem`
+- `pxe`: creates an image that can be booted via PXE (network boot), see
+  :ref:`build_pxe`
+
+- `docker`, `oci`: container images, see :ref:`building-docker-build`
+
+- `btrfs`, `ext2`, `ext3`, `ext4`, `xfs`: KIWI will convert the
+  image into a mountable filesystem of the specified type.
+
+- `squashfs`, `clicfs`: creates the image as a filesystem that can be used
+  on live systems
+
+- `tbz`, `cpio`: the unpacked source tree will be compressed into a `XZ
+  <https://en.wikipedia.org/wiki/Xz>`_ or `CPIO
+  <https://en.wikipedia.org/wiki/Cpio>`_ archive.
+
+.. TODO: explain subelements like <vagrantconfig>
+   TODO: explain the `primary` attribute
+
+.. _xml-description-preferences-common-elements:
+
+Common elements
+^^^^^^^^^^^^^^^
+
+
+.. _xml-description-image-profiles:
+
+Image profiles
+--------------
+
+
+Adding users
+------------
+
+User accounts can be added or modified via the `users` element, which
+supports a list of multiple `user` sub-elements:
+
+.. code-block:: xml
+
+   <?xml version="1.0" encoding="utf-8"?>
+
+   <image schemaversion="6.9" name="JeOS-Tumbleweed">
+     <users>
+       <user password="this_is_soo_secure" home="/home/me" name="me" groups="users" pwdformat="plain" />
+       <user password="$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0" home="/root" name="root" groups="root"/>
+     </users>
+
+   </image>
+
+Each `user` element represents a specific user that should be added or
+modified. The following attributes are mandatory:
+
+- `name`: the UNIX username
+
+- `home`: the path to the user's home directory
+
+Additionally the following optional attributes can be specified:
+
+- `groups`: A comma separated list of groups. The first element of the list
+  is used for the user's primary group. The remaining elements are appended
+  to the user's supplementary groups. When no groups are assigned then the
+  system's default primary group will be used [#f2]_
+
+- `id`: The user-id of this account
+
+- `pwdformat`: The format in which `password` is provided, either `plain`
+  or `encrypted` (the latter is the default)
+
+- `password`: The password for this user account. It can be provided either
+  in cleartext form (`pwdformat="plain"`) or in `crypt`'ed form
+  (`pwdformat="encrypted"`). Plain passwords are discouraged, as everyone
+  with access to the image description would know the password. Thus please
+  generate a hash of your password, e.g. with the `mkpasswd` tool
+  (available in most Linux distributions via the `whois` package):
+
+  .. code:: bash
+
+     $ mkpasswd -m sha-512 -S $(date +%N) -s <<< INSERT_YOUR_PASSWORD_HERE
+
+
+Defining repositories and packages
+----------------------------------
+
```

Comment:
Is there are paragraph missing? It looks like an empty section to me.

Reply:
Yes, I haven't written that one yet. That's why I've marked this as a draft & "work in progress".

Reply:
Right. :laughing: